dspace/config/spring/rest/matomo.xml

```diff
@@ -0,0 +1,53 @@
+        <constructor-arg name="enrichers">
+            <list>
+                <bean id="userAgentEnricher" class="org.dspace.matomo.factory.MatomoRequestDetailsEnricherFactory" factory-method="userAgentEnricher"/>
+                <bean id="actionNameEnricher" class="org.dspace.matomo.factory.MatomoRequestDetailsEnricherFactory" factory-method="actionNameEnricher"/>
+                <bean id="urlEnricher" class="org.dspace.matomo.factory.MatomoRequestDetailsEnricherFactory" factory-method="urlEnricher"/>
+                <bean id="downloadEnricher" class="org.dspace.matomo.factory.MatomoRequestDetailsEnricherFactory" factory-method="downloadEnricher"/>
+                <bean id="cookieIdEnricher" class="org.dspace.matomo.factory.MatomoRequestCookieIdentifierEnricher"/>
+                <bean id="cookieSessionEnricher" class="org.dspace.matomo.factory.MatomoRequestCookieSessionEnricher"/>
+                <bean id="customVariablesEnricher" class="org.dspace.matomo.factory.MatomoRequestCustomVariablesEnricher"/>
+                <bean id="customCookiesEnricher" class="org.dspace.matomo.factory.MatomoRequestCustomCookiesEnricher">
+                        <constructor-arg name="customCookies" value="${matomo.request.customcookies.cookie-name:}"/>
+                </bean>
+                <!-- These two enrichers requires a valid token set on the client -->
+<!--                <bean id="ipEnricher" class="org.dspace.matomo.factory.MatomoRequestIpAddressEnricher"/>-->
+<!--                <bean id="requestCountryEnricher" class="org.dspace.matomo.factory.MatomoRequestCountryEnricher"/>-->
+            </list>
+        </constructor-arg>
+        <constructor-arg name="siteId" value="${matomo.request.siteid:}"/>
+    </bean>
+
+    <bean id="matomoDefaultClient" class="org.dspace.matomo.client.MatomoClientImpl">
+        <constructor-arg name="baseUrl" value="${matomo.default-client.baseurl:}"/>
+        <constructor-arg name="token" value="${matomo.default-client.token:}"/>
+        <constructor-arg name="matomoRequestBuilder" ref="matomoRequestBuilder"/>
+        <constructor-arg name="matomoResponseReader" ref="matomoResponseReader"/>
+        <constructor-arg name="httpClient">
+            <bean class="org.apache.http.impl.client.HttpClients" factory-method="createDefault"/>
+        </constructor-arg>
+    </bean>
+
+    <bean id="matomoAsyncClient" class="org.dspace.matomo.client.MatomoAsyncClientImpl">
+        <constructor-arg name="baseUrl" value="${matomo.async-client.baseurl:}"/>
+        <constructor-arg name="token" value="${matomo.async-client.token:}"/>
+        <constructor-arg name="matomoRequestBuilder" ref="matomoRequestBuilder"/>
+        <constructor-arg name="matomoResponseReader" ref="matomoResponseReader"/>
+    </bean>
+
+    <bean id="matomoSyncEventHandler" class="org.dspace.matomo.MatomoSyncEventHandler">
+        <constructor-arg name="matomoClient" ref="matomoAsyncClient"/>
```

Comment:
It would help to have documentation for this file. I'm unable to understand why, for for a sync event handler we have the AsyncClient. Why can't we use the default client? Is it intend to have different clients here, could he have all options listed here (but under comments)?

Reply:
The event handler has its own business logic, collects events and then sends them to the `matomoClient` bean configured, in order to communicate those events to our tracking service.
This client can have its own business logic, so should not be coupled with the event-handler.
With this approach you can use all the combinations you prefer, depending also on your infrastructure.

Reply:
Thank you @vins01-4science for addressing my feedback!